core/consensus/babe/primitives/src/lib.rs

```diff
@@ -133,6 +138,90 @@ impl slots::SlotData for BabeConfiguration {
 	const SLOT_KEY: &'static [u8] = b"babe_bootstrap_data";
 }
 
+/// Represents an Babe equivocation proof.
+#[derive(Debug, Clone, Encode, Decode, PartialEq)]
+pub struct BabeEquivocationProof<H> {
```

Comment:
I thought we had a type for `EquivocationProof` already (in slots). Why add another? I also don't know why you need the authority signatures here, if you pass the headers you can extract the author from it.

Reply:
Because we are taking the signatures out of the header (I think we talk about this yesterday). In your case I will need to push the signature again, and pop it again when checking...

Reply:
I don't think that is possible unless we keep the signature inside. I think signature outside is more flexible.